signalfx-metrics/src/main/java/com/signalfx/metrics/aws/AWSInstanceInfo.java

```diff
@@ -41,18 +43,20 @@ public static String get(int timeoutInMs) {
 
         try {
             HttpResponse response = client.execute(request);
-            JsonNode object = MAPPER.readTree(response.getEntity().getContent());
-
-            return object.get(INSTANCE_ID).asText() + "_" + object.get(REGION).asText() + "_"
-                    + object.get(ACCOUNT_ID).asText();
-
+            return parse(response.getEntity().getContent());
         } catch (Exception e) {
             log.trace("Exception trying to execute {}, Exception: {} ", request, e);
         }
 
         return null;
     }
 
+    @VisibleForTesting
+    static String parse(InputStream inputStream) throws Exception {
+        Map<String, Object> result = JSON.std.mapFrom(new InputStreamReader(inputStream, StandardCharsets.UTF_8));
```

Comment:
Shouldn't we close the `inputStream`?

Reply:
Thanks, I'll change the caller to close the stream.